app/domain/authentication/authn_azure/validate_application_identity.rb

```diff
@@ -153,12 +155,20 @@ def validate_token_identity_matches_annotations
           annotation_type  = constraint[0].to_s
           annotation_value = constraint[1]
           unless annotation_value == @token_identity[annotation_type.to_sym]
-            raise Err::InvalidApplicationIdentity.new(annotation_type)
+            raise Err::InvalidApplicationIdentity.new(annotation_type_constraint(annotation_type))
           end
         end
         @logger.debug(Log::ValidatedApplicationIdentity.new)
       end
 
+      def annotation_type_constraints constraints
+        constraints.map { |constraint| annotation_type_constraint(constraint) }
+      end
+
+      def annotation_type_constraint constraint
```

Comment:
I think we should add comment to the function that explain why we are doing it

Reply:
Thanks